src/lib/comms/sol-lwm2m.c

```diff
@@ -4718,6 +4859,172 @@ client_bootstrap(void *data)
+    uint16_t target_object_id,
+    uint16_t target_instance_id,
+    int64_t owner_server_id,
+    struct sol_lwm2m_resource *acl_res,
+    bool register_with_coap)
+{
+    struct obj_ctx *acc_obj_ctx;
+    int r;
+    struct sol_buffer buf = SOL_BUFFER_INIT_FLAGS(NULL, 0, SOL_BUFFER_FLAGS_NO_NUL_BYTE);
+    struct sol_lwm2m_resource *res;
+    struct sol_lwm2m_payload payload = { .type = SOL_LWM2M_CONTENT_TYPE_TLV };
+
+    acc_obj_ctx = find_object_ctx_by_id(client, ACCESS_CONTROL_OBJECT_ID);
+
+    //Only the Client itself and the owner server is able to manage
+    // Access Control Object Instances, and Security Object Instances
+    // can only be managed by Bootstrap Servers, so there's no sense in
+    // creating Access Control Object Instances or ACLs for them
+    if (target_object_id == SECURITY_SERVER_OBJECT_ID ||
+        target_object_id == ACCESS_CONTROL_OBJECT_ID) {
+        if (acl_res)
+            sol_lwm2m_resource_clear(acl_res);
+        return 0;
+    }
+
+    if (acl_res) {
+        res = calloc(4, sizeof(struct sol_lwm2m_resource));
+        res[2] = *acl_res;
+    } else
+        res = calloc(3, sizeof(struct sol_lwm2m_resource));
+
+    SOL_LWM2M_RESOURCE_INT_INIT(r, &res[0],
+        ACCESS_CONTROL_OBJECT_OBJECT_RES_ID, target_object_id);
+    if (r < 0) {
+        SOL_WRN("Could not init Access Control Object's [Object ID] resource\n");
+        return r;
+    }
+    SOL_LWM2M_RESOURCE_INT_INIT(r, &res[1],
+        ACCESS_CONTROL_OBJECT_INSTANCE_RES_ID, target_instance_id);
+    if (r < 0) {
+        SOL_WRN("Could not init Access Control Object's [Instance ID] resource\n");
+        return r;
+    }
+    SOL_LWM2M_RESOURCE_INT_INIT(r, &res[acl_res ? 3 : 2],
+        ACCESS_CONTROL_OBJECT_OWNER_RES_ID, owner_server_id);
+    if (r < 0) {
+        SOL_WRN("Could not init Access Control Object's [Owner ID] resource\n");
+        return r;
+    }
+
+    //From array of sol_lwm2m_resource to a sol_buffer in TLV format
+    r = resources_to_tlv(res, acl_res ? 4 : 3, &buf);
+    clear_resource_array(res, (acl_res ? 4 : 3));
+    free(res);
+    SOL_INT_CHECK(r, < 0, r);
+
+    //From sol_buffer to a sol_vector of sol_lwm2m_tlv
+    r = sol_lwm2m_parse_tlv(sol_buffer_get_slice(&buf), &payload.payload.tlv_content);
+    sol_buffer_fini(&buf);
+    SOL_INT_CHECK(r, < 0, r);
+
+    r = handle_create(client, acc_obj_ctx, -1, payload, UINT16_MAX, register_with_coap);
+    sol_lwm2m_tlv_list_clear(&payload.payload.tlv_content);
+    if (r == SOL_COAP_RESPONSE_CODE_CHANGED || r == SOL_COAP_RESPONSE_CODE_CREATED) {
+        SOL_DBG("Access Control Object Instance for Object Instance /%"
+            PRIu16 "/%" PRIu16 " created successfully!",
+            target_object_id, target_instance_id);
+    } else {
+        SOL_WRN("Failed to create Access Control Object Instance for Object Instance /%"
+            PRIu16 "/%" PRIu16, target_object_id, target_instance_id);
+        return -ENOENT;
+    }
+
+    return r;
```

Comment:
I need the response code in the caller function. From there I `return 0;` already.